src/test/scala/firrtlTests/CheckSpec.scala

```diff
@@ -536,6 +566,20 @@ class CheckSpec extends AnyFlatSpec with Matchers {
     }
   }
 
+  it should "throw an exception if a statement name is used as a reference" in {
+    val src = """
+                |circuit test:
+                |  module test:
+                |    input clock: Clock
+                |    output a: UInt<2>
+                |    stop(clock, UInt(1), 1) : hello
+                |    a <= hello
+                |""".stripMargin
+    assertThrows[CheckHighForm.UndeclaredReferenceException] {
+      checkHighInput(src)
+    }
```

Comment:
Should we provide a more specific error for this case?

Reply:
I think this should be good enough for now. Very few people actually write firrtl and when you use Chisel, this error case should never happen.